pkg/build/api/v1beta3/types.go

```diff
@@ -382,6 +386,21 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// BuildPostCommitSpec holds a build post commit hook specification.
+type BuildPostCommitSpec struct {
+	// Shell is an action that executes a shell script in a container
+	// running the build output image.
+	Shell string `json:"shell,omitempty" description:"shell script to be executed in a container running the build output image"`
```

Comment:
The positional args thing I think is what gets me over the hump - it's the closest to what I want and it's supported in bash - `/bin/bash -c STRING -- ARG0 ARG1`, passed as an array of argument into the container launch command (so that we preserve strong argument separation).